scripts/build/gen_relocate_app.py

```diff
@@ -605,15 +614,33 @@ def main():
     generate_linker_script(linker_file, sram_data_linker_file,
                            sram_bss_linker_file, complete_list_of_sections, phdrs)
 
+    delayed_relocate = dict()
+    if os.path.isfile(args.edt_pickle):
+        with open(args.edt_pickle, 'rb') as f:
+            edt = pickle.load(f)
+            for node in edt.compat2nodes["zephyr,memory-region"]:
+                delayed_relocate[node.props["zephyr,memory-region"].val] = node.props["delay-relocation"].val
+
+    else:
+        edt = None
```

Comment:
If you really want to handle the file not existing, it would be better instead to handle `FileNotFoundError` exceptions from `open()` so that there is no change of a race condition between the check and the use of the file.

But I think it's wrong to set `edt = None` if the file is not found. It should be set to `None` only if the `args.edt_pickle is None`.